test/unit/WamplOracle.ts

```diff
@@ -0,0 +1,137 @@
+import { ethers, waffle } from 'hardhat'
+import { BigNumber, Contract, Signer } from 'ethers'
+import { expect } from 'chai'
+
+let mockAmplEthAggregator: Contract
+let mockEthUsdAggregator: Contract
+let mockAmpl: Contract
+let mockWampl: Contract
+let oracle: Contract
+let deployer: Signer
+let user: Signer
+
+async function mockedOracle() {
+  const [deployer, user] = await ethers.getSigners()
+  // deploy mocks
+  const mockAmplEthAggregator = await (
+    await ethers.getContractFactory('MockChainlinkAggregator')
+  )
+    .connect(deployer)
+    .deploy(18)
+  const mockEthUsdAggregator = await (
+    await ethers.getContractFactory('MockChainlinkAggregator')
+  )
+    .connect(deployer)
+    .deploy(8)
+  const mockAmpl = await (await ethers.getContractFactory('MockAMPL'))
+    .connect(deployer)
+    .deploy(9)
+  const mockWampl = await (await ethers.getContractFactory('MockWAMPL'))
+    .connect(deployer)
+    .deploy(18, mockAmpl.address)
+  // deploy contract to test
+  const oracle = await (await ethers.getContractFactory('WamplOracle'))
+    .connect(deployer)
+    .deploy(
+      mockAmplEthAggregator.address,
+      mockEthUsdAggregator.address,
+      mockWampl.address,
+      60000,
+    )
+
+  return {
+    deployer,
+    user,
+    mockAmplEthAggregator,
+    mockEthUsdAggregator,
+    mockAmpl,
+    mockWampl,
+    oracle,
+  }
+}
+
+describe('WamplOracle', function () {
+  before('setup Orchestrator contract', async () => {
+    ;({
+      deployer,
+      user,
+      mockAmplEthAggregator,
+      mockEthUsdAggregator,
+      mockAmpl,
+      mockWampl,
+      oracle,
+    } = await waffle.loadFixture(mockedOracle))
+  })
+
+  describe('when sent ether', async function () {
+    it('should reject', async function () {
+      await expect(user.sendTransaction({ to: oracle.address, value: 1 })).to.be
+        .reverted
+    })
+  })
+
+  describe('Fetching data', async function () {
+    it('should succeed with fresh data', async function () {
```

Comment:
think we could use some more tests -
- various wrapperToUnderlying values
- Either oracle returns some wonky super high or super low value
- gas test so we can optimize a bit